src/Core.tsx

```diff
@@ -221,7 +221,19 @@ export class Core extends React.Component<IModalProps, IModalState> {
     return this.validateCurrentChain()
   }
 
-  private continueSettingUp = (provider: any) => this.setupProvider(provider).then((success) => { if (success) { return this.detectFlavor() } })
+  /**
+   * ContinueSettingUp
+   * After connecting to the provider but before detecting the flavor
+   */
+  private continueSettingUp = (provider: any) => this.setupProvider(provider).then((success) => {
+    if (provider.chooseAccount) {
```

Comment:
I would also add a "whitelist" for the providers that we support for choosing accounts. We can add it _lib/hardware-wallets_. Then, also check it has the expected method